### ISSUE TITLE
SPARQL for NaN values fails

### ISSUE THREAD

**Issue by joernhees** [openlink/virtuoso-opensource#649]
i think there's a bug in the query parser when trying to query for NaN values:
```
select ?s where {
  ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> .
}
limit 10
```

try on [dbpedia](http://dbpedia.org/sparql?default-graph-uri=http%3A%2F%2Fdbpedia.org&qtxt=select+%3Fs+where+%7B%0D%0A++%3Fs+%3Fp+%22NaN%22%5E%5E%3Chttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23float%3E+.%0D%0A%7D%0D%0Alimit+10&format=text%2Fhtml&CXML_redir_for_subjs=121&CXML_redir_for_hrefs=&timeout=30000&debug=on)

```
Virtuoso 22005 Error SR334: Invalid floating point value converting 'NaN'

SPARQL query:
define sql:big-data-const 0 
#output-format:text/html
define sql:signal-void-variables 1 define input:default-graph-uri <http://dbpedia.org> select ?s where {
  ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> .
}
limit 10
```

**Comment by joernhees** [comment]
same seems true for `INF`, both also as implicit and as float & double ... implicitly like `?s ?p NaN` or `?s ?p INF` i get other weird errors though...

**Comment by HughWilliams** [comment]
@joernhees: I have been able to recreate even with an empty database:

```
SQL> SPARQL select ?s where {   ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> . } limit 10;

*** Error 37000: VD [Virtuoso Server]SQ074: Line 1: SR334: Invalid floating point value converting 'NaN'
at line 15 of Top-Level:
SPARQL select ?s where {   ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> . } limit 10;select ?s where {   ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> . } limit 10
SQL> status('');
REPORT
VARCHAR
_______________________________________________________________________________

OpenLink Virtuoso VDB Server
Version 07.20.3219-pthreads for Darwin as of Apr 25 2017
```

 and reported to development to look into ...

**Comment by HughWilliams** [comment]
@joernhees This issue has been resolved in the latest Virtuoso 7.2.5 open source stable/7 release made in Aug 2018 , see http://vos.openlinksw.com/owiki/wiki/VOS/

```
SQL> SPARQL select ?s where {   ?s ?p "NaN"^^<http://www.w3.org/2001/XMLSchema#float> . } limit 10;
s
LONG VARCHAR
_______________________________________________________________________________


0 Rows. -- 1 msec.
SQL> 
```
